CMakeLists.txt

```diff
@@ -145,6 +145,13 @@ if(TRITON_BUILD_PYTHON_MODULE)
     add_link_options(${Python3_LINK_OPTIONS})
   endif()
 
+  if (DEFINED TRITON_EXTERNAL_CODEGEN_BACKEND_DIRS)
```

Comment:
I'd like to suggest we rename this "TRITON_PLUGIN_DIRS". 

Referring to external code as "plugins" in general in the source, docs, etc. sounds better (to me at least) and more aligns with how the industry talks about this type of code.